lib/api/model/submessage.dart

```diff
@@ -313,3 +318,129 @@ class UnknownPollEventSubmessage extends PollEventSubmessage {
   @override
   Map<String, Object?> toJson() => json;
 }
+
+/// States of a poll Zulip widget.
+///
+/// See also:
+/// - https://zulip.com/help/create-a-poll
+/// - https://github.com/zulip/zulip/blob/304d948416465c1a085122af5d752f03d6797003/web/shared/src/poll_data.ts
+class Poll {
+  Poll({
+    required this.pollSenderId,
+    required this.question,
+    required final List<String> options,
+  }) {
+    for (int index = 0; index < options.length; index += 1) {
+      // Initial poll options use a placeholder senderId.
+      // See [PollEventSubmessage.optionKey] for details.
+      _addOption(null, options[index], idx: index);
+    }
+  }
+
+  factory Poll.fromSubmessages(
+    List<Submessage> submessages,
+    {required int senderId, required PollWidgetData widgetData}
+  ) {
```

Comment:
Because finding out which widget the message contains requires the caller to parse `WidgetData` first, they are asked to include `widgetData` just to avoid parsing it.

`senderId` is the sender ID of the enclosing message.

Reply:
I think a way to make it clearer is to ask the caller to include only the supposed PollEventSubmessage, by skipping the first submessage on the message, and make them all named:

```suggestion
  factory Poll.fromSubmessages({
    required Iterable<Submessage> pollEventSubmessages,
    required int messageSenderId,
    required PollWidgetData widgetData
  }) {
```

Reply:
Sounds good.

Let's order them logically, too: message sender ID first since the message encloses the whole thing, and widget data before event submessages.

Reply:
> it seems like it's assumed that it includes the [PollWidgetData]. Probably cleaner to have it not include that.

(And because I see this was ambiguous: what I intended here was to have the `submessages` argument not include the [PollWidgetData].)